src/Plugin/PasswordConstraint/PasswordConstraintBase.php

```diff
@@ -0,0 +1,151 @@
+<?php
+
+namespace Drupal\password_enhancements\Plugin\PasswordConstraint;
+
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\Plugin\PluginBase;
+use Drupal\password_enhancements\PasswordConstraintInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides a base class for password constraints.
+ *
+ * @see \Drupal\password_enhancements\Annotation\PasswordConstraint
+ * @see \Drupal\password_enhancements\PasswordConstraintInterface
+ * @see \Drupal\password_enhancements\PasswordConstraintManager
+ * @see plugin_api
+ */
+abstract class PasswordConstraintBase extends PluginBase implements PasswordConstraintInterface, ContainerFactoryPluginInterface {
+
+  /**
+   * The password constraint ID.
+   *
+   * @var string
+   */
+  protected $uuid;
+
+  /**
+   * PasswordConstraintBase constructor.
+   *
+   * @param array $configuration
+   *   The configuration array.
+   * @param string $plugin_id
+   *   The plugin ID.
+   * @param mixed $plugin_definition
+   *   The plugin definition.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition) {
+    $this->uuid = $configuration['uuid'] ?? '';
```

Comment:
What if you would DI the UUID generator service to this class and properly initialize the missing UUID here instead of setting it to an empty string?

Reply:
https://www.previousnext.com.au/blog/safely-extending-drupal-8-plugin-classes-without-fear-of-constructor-changes

Reply:
There's a problem with that: `\Drupal\Core\Plugin\PluginBase` doesn't have a `create()` method, and the UUID generator service should be available right within the constructor.

Reply:
`Drupal\Core\Plugin\ContainerFactoryPluginInterface`

https://www.drupal.org/docs/8/api/services-and-dependency-injection/dependency-injection-in-plugin-block